controllers/djob_controller.go

```diff
@@ -92,12 +92,12 @@ func (r *DjobReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	}
 
 	if instance.IsSubmitted() {
-		r.Log.Info(fmt.Sprintf("Refresh for %v", req.NamespacedName))
-		if err := r.refresh(instance); err != nil {
-			r.Recorder.Event(instance, corev1.EventTypeWarning, "Refreshing object", fmt.Sprintf("Failed to refresh object: %s", err))
-			return ctrl.Result{}, fmt.Errorf("error when refreshing job: %v", err)
+		r.Log.Info(fmt.Sprintf("Reset for %v", req.NamespacedName))
+		if err := r.reset(instance); err != nil {
+			r.Recorder.Event(instance, corev1.EventTypeWarning, "Resetting object", fmt.Sprintf("Failed to reset object: %s", err))
+			return ctrl.Result{}, fmt.Errorf("error when resetting job: %v", err)
 		}
-		r.Recorder.Event(instance, corev1.EventTypeNormal, "Refreshed", "Object is refreshed")
+		r.Recorder.Event(instance, corev1.EventTypeNormal, "Reseted", "Object is reseted")
 	}
 
 	return ctrl.Result{RequeueAfter: 30 * time.Second}, nil
```

Comment:
I think we need both Reset and Refresh, as you can see there is a re queue every 30 seconds  and we don't want to reset job every 30 seconds.  You might need to check if the job settings has changed/updated and based on that reset or refresh

Reply:
@Azadehkhojandi  Actually Refresh here it always giving me an error, however, when I skipped the error and goes to  r.Update(context.Background(), instance), it doesn't make any change in the object. why do we need the update here?

Reply:
done